src/backend/joanie/core/models/products.py

```diff
@@ -965,9 +967,25 @@ def submit_for_signature(self, user: User):
         )
 
         if should_be_resubmitted:
-            backend_signature.delete_signing_procedure(
-                contract.signature_backend_reference
-            )
+            # The signature provider may take some time to respond with the confirmation of the
+            # deletion. If we reach the timeout limit, we should reset the contract submission
+            # values because the signature provider will delete them.
+            # We won't be able to use the `contract.signature_backend_reference` again.
+            # There can be edge case where the signature backend reference was already deleted,
+            # causing an error that the reference was not found at the signature provider side.
+            try:
+                backend_signature.delete_signing_procedure(
+                    contract.signature_backend_reference
+                )
+            except BackendTimeout as exception:  # pylint: disable=unused-variable
+                handle_signature_deletion_error(
+                    order=self, error_message="Timeout on signature reference deletion"
+                )
+            except DeleteSignatureProcedureFailed as exception:  # pylint: disable=unused-variable
+                handle_signature_deletion_error(
+                    order=self, error_message="Failed to delete signature procedure"
+                )
```

Comment:
```suggestion
            except (BackendTimeout, DeleteSignatureProcedureFailed) as exception:
                logger.error(
                    exception,
                    extra={
                        "context": {
                            "order": self.to_dict(),
                            "product": self.product.to_dict(),
                            "signature_backend_reference": (
                                contract.signature_backend_reference
                            ),
                        }
                    },
                )
                contract.reset_submission_for_signature()
            was_already_submitted = False
```
IMO it's ok like this.